src/ctapipe/tools/optimize_event_selection.py

```diff
@@ -0,0 +1,234 @@
+
+    aliases = {
+        "gamma-file": "EventSelectionOptimizer.gamma_file",
+        "proton-file": "EventSelectionOptimizer.proton_file",
+        "electron-file": "EventSelectionOptimizer.electron_file",
+        "output": "EventSelectionOptimizer.output_path",
+        "chunk_size": "EventSelectionOptimizer.chunk_size",
+    }
+
+    classes = [EventLoader] + classes_with_traits(CutOptimizerBase)
+
+    def setup(self):
+        """
+        Initialize components from config.
+        """
+        self.optimizer = CutOptimizerBase.from_name(
+            self.optimization_algorithm, parent=self
+        )
+        self.particles = [
+            EventLoader(
+                parent=self,
+                kind="gammas",
+                file=self.gamma_file,
+                target_spectrum=self.gamma_target_spectrum,
+            )
+        ]
+        if not isinstance(self.optimizer, PercentileCuts):
+            if not self.proton_file or (
+                self.proton_file and not self.proton_file.exists()
+            ):
+                raise ValueError(
+                    "Need a proton file for cut optimization "
+                    f"using {self.optimization_algorithm}."
+                )
+
+            self.particles.append(
+                EventLoader(
+                    parent=self,
+                    kind="protons",
+                    file=self.proton_file,
+                    target_spectrum=self.proton_target_spectrum,
+                )
+            )
+            if self.electron_file and self.electron_file.exists():
+                self.particles.append(
+                    EventLoader(
+                        parent=self,
+                        kind="electrons",
+                        file=self.electron_file,
+                        target_spectrum=self.electron_target_spectrum,
+                    )
+                )
+            else:
+                self.log.warning("Optimizing cuts without electron file.")
+
+    def start(self):
+        """
+        Load events and optimize g/h (and theta) cuts.
+        """
+        reduced_events = dict()
+        for sel in self.particles:
+            evs, cnt, meta = sel.load_preselected_events(self.chunk_size, self.obs_time)
+            if isinstance(self.optimizer, PointSourceSensitivityOptimizer):
+                evs = sel.make_event_weights(
+                    evs,
+                    meta["spectrum"],
+                    (
+                        self.optimizer.min_bkg_fov_offset,
+                        self.optimizer.max_bkg_fov_offset,
+                    ),
+                )
+
+            reduced_events[sel.kind] = evs
+            reduced_events[f"{sel.kind}_count"] = cnt
+            if sel.kind == "gammas":
+                self.sim_info = meta["sim_info"]
+                self.gamma_spectrum = meta["spectrum"]
+
+        self.signal_events = reduced_events["gammas"]
+
+        if isinstance(self.optimizer, PercentileCuts):
```

Comment:
Same comment here. Could be implemented via a `requires_background` property or similar.

Reply:
I combined both these properties into one